File: EyeTurner/app/src/main/java/raymondbdev/eyeturner/Model/ParentViewModel.java
package raymondbdev.eyeturner.Model;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.github.mertakdut.Reader;

public class ParentViewModel extends ViewModel {
    private final MutableLiveData<GazeTrackerHelper> mutableGazeTrackerHelper =
            new MutableLiveData<>();



    public void setTracker(GazeTrackerHelper item) {
        mutableGazeTrackerHelper.setValue(item);
    }

    public LiveData<GazeTrackerHelper> getTracker() {
        return mutableGazeTrackerHelper;
    }

    //    private final MutableLiveData<Integer> mutablePageIndex = new MutableLiveData<>();
//    private final MutableLiveData<Reader> mutableBookReader = new MutableLiveData<>();

//    public void setBookReader(Reader item) {
//        mutableBookReader.setValue(item);
//    }
//
//    public LiveData<Reader> getBookReader() {
//        return mutableBookReader;
//    }
//
//    public void setPageIndex(Integer item) {
//        mutablePageIndex.setValue(item);
//    }
//
//    public LiveData<Integer> getPageIndex() {
//        return mutablePageIndex;
//    }
}

File: EyeTurner/app/src/main/java/raymondbdev/eyeturner/Model/ParentViewModel.kt
package raymondbdev.eyeturner.Model

import android.content.ContentResolver
import android.os.Build
import android.os.VibrationEffect
import android.os.Vibrator
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

/**
 * Wrapper Class which allows us to access persistive objects between fragments.
 */
class ParentViewModel : ViewModel() {
    private val mutableGazeTrackerHelper = MutableLiveData<GazeTrackerHelper>()
    private val mutableSettingsManager = MutableLiveData<SettingsManager>()
    private val mutableReadingTracker = MutableLiveData<ReadingTracker>()
    private val mutableVibrator = MutableLiveData<Vibrator>()
    private val mutableContentResolver = MutableLiveData<ContentResolver>()

    val tracker: GazeTrackerHelper?
        get() = mutableGazeTrackerHelper.value

    val settingsManager: SettingsManager?
        get() = mutableSettingsManager.value

    val readingTracker: ReadingTracker?
        get() = mutableReadingTracker.value

    fun setTracker(helper: GazeTrackerHelper) {
        mutableGazeTrackerHelper.value = helper
    }

    fun setSettingsManager(manager: SettingsManager) {
        mutableSettingsManager.value = manager
    }

    fun setVibrator(vibrator: Vibrator) {
        mutableVibrator.value = vibrator
    }

    fun setMutableContentResolver(contentResolver: ContentResolver) {
        mutableContentResolver.value = contentResolver
    }

    fun setReadingHelper(helper: ReadingTracker) {
        mutableReadingTracker.value = helper
    }

    fun vibrate() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) { // vibrate API only available
            mutableVibrator.value!!.vibrate(VibrationEffect.createPredefined(VibrationEffect.EFFECT_CLICK))
        }
    }

    fun getMutableContentResolver(): ContentResolver? {
        return mutableContentResolver.value
    }


}
